content/docs/guides/ha_scale/perf-scale.md

```diff
@@ -0,0 +1,121 @@
+---
+title: "Performance and Scalability"
+description: "Performance benchmark on control plane and data plane sidecar and scalability test of Open Service Mesh"
+type: docs
+---
+
+Open Service Mesh aims to create a high-performance service mesh while providing all the essential service mesh features. This document describes the tests performed to evaluate the performance and scalability of OSM.
+
+## Testing environment
+
+This test is performed on a Kubernetes cluster provisioned and managed by [Microsoft Azure Kubernetes Service](https://azure.microsoft.com/en-us/services/kubernetes-service/). The cluster has 100 Standard_DS2_v2 nodes (each node has 2 vCPUs and 7GB memory). It is ensured that there is only one OSM controller pod running with default resource quota during the test. OSM is configured in non-permissive traffic mode.
+
+## Process
+
+The testing mesh contains two types of service. One is a load generator (modified from [wrk2](https://github.com/giltene/wrk2)), which can send requests at specified rate, load balanced among all other services in the mesh. The load generator serves as the client side of a request. Throughout our test, there will be at most one load generator service. The other type of service is a simple echo application that replies with whatever it receives from an inbound request. It serves as the server side of a request. We can deploy multiple copies of the echo application in the mesh. In short, the mesh topology is a typical 1-to-N service architecture.
+
+<p align="center">
+  <img src="/docs/images/perf/mesh-topo.png" width="650"/>
+</p>
+
+
+## Performance
+
+In this test, we focus on the latency overhead and resource consumption added by OSM's. We currently test the per pod RPS at 100, 500, 1000, 2000 and 4000 respectively.
+
+### Latency
+
+Below is the request latency at different RPS levels with and without OSM installed. Note, with OSM installed the added latency comes from both the client and server sidecar.
+
+<p align="center">
+  <img src="/docs/images/perf/wo-osm-latency.png" width="650"/>
+</p>
+
+<p align="center">
+  <img src="/docs/images/perf/w-osm-latency.png" width="650"/>
+</p>
+
+<table style="width: 100%">
+<thead>
+<tr>
+<th>Per pod RPS</th>
+<th>100</th>
+<th>500</th>
+<th>1000</th>
+<th>1500</th>
+<th>2000</th>
+<th>4000</th>
+</tr>
+</thead>
+<tbody>
+<tr>
+<td>P99 latency without OSM</td>
+<td>2</td>
+<td>4.6</td>
+<td>3.7</td>
+<td>4.23</td>
+<td>4.3</td>
+<td>3.1</td>
+</tr>
+<tr>
+<td>P99 latency with OSM</td>
+<td>10.1</td>
```

Comment:
why is 100 rps so much higher than 500 rps? can we try running this a few times?

Reply:
are we appropriately chopping off the beginning metrics of the test to account for a stable OSM state?